### PR TITLE
feat(cannon): Proof PreimageOffset Field

### PR DESCRIPTION
**Pull request by refcell**
**Description**

This PR adds a new `PreimageOffset` field to cannon's
`Proof` object which is required for the challenger
to construct valid preimage oracle update calls.

**Metadata**

Fixes CLI-4320

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6588** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6588" target="_blank">Graphite</a>  👈
    * **PR #6571** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6571" target="_blank">Graphite</a> 
      * **PR #6572** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6572" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6588?utm_source=stack-comment).

File: cannon/README.md [hunk 1]
```diff
@@ -39,25 +39,24 @@ make cannon
...
---l2.claim 0x530658ab1b1b3ff4829731fc8d5955f0e6b8410db2cd65b572067ba58df1f2b9
---l2.blocknumber 8813570
---datadir /tmp/fpp-database
---server
+./bin/cannon run \
+  --pprof.cpu \
+  --info-at '%10000000' \
+  --proof-at never \
+  --input ./state.json \
+  -- ../op-program/bin/op-program \
```
Comment (ajsutton) [inline cannon/README.md]: I'd probably undo this change and go back to having the `--` on a line by itself.  It's an important detail that is much easier to see when it's on its own line.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: cannon/README.md (hunk 1 above)
Comment (refcell) [reply cannon/README.md]: Hmm bash requires a backslash at the end of the line to make it a valid multiline command. This makes a copy-paste of this not work in bash. I will revert, but much easier to have this to allow copy + pasting the command.

Comment (Inphi) [reply cannon/README.md]: you should keep the backslash, but retain `-- \` as a separate line.

Comment (refcell) [reply cannon/README.md]: Ah got it, already removed but I will follow up with the change in a new pr
